### PR TITLE
Code restructuring 1

### PR DESCRIPTION
**Pull request by matheper**
- Moved project.modular_llm utils and evaluators to mttl. Some code was merged to mttl.utils, mttl.callbacks, and mttl.evaluators.
- Created mttl.logging
- Removed some unused code

### CODE REVIEW

File: mttl/models/expert_trainer.py [hunk 1]
```diff
@@ -10,15 +7,8 @@
 from transformers.trainer import Trainer
 from transformers.utils import PushToHubMixin
 
+from mttl.logging import logger
 from mttl.models.get_scheduler import get_scheduler_with_args
-from mttl.utils import logger
-
-
-def transfer_batch_to_device(batch, device):
```
Comment (sordonia) [inline mttl/models/expert_trainer.py:17]: this was unused?

File: mttl/utils.py [hunk 2]
```diff
@@ -555,3 +198,18 @@ def wrapped_fn(*args, **kwargs):
...
 def generate_random_string(str_len=10):
     return "".join(random.choices(string.ascii_uppercase, k=str_len))
+
+
+# from projects.modular_llm.src.utils.evaluators import Evaluator
+def get_loss(model, evaluator: "Evaluator", **kwargs):
+    return evaluator.get_loss(model, **kwargs)
+
+
+def get_svd_embedding(lib, expert_name: str):
```
Comment (sordonia) [inline mttl/utils.py:208]: maybe in library? mttl/utils.py seems a bit too generic

File: mttl/utils.py [hunk 3]
```diff
@@ -555,3 +198,18 @@ def wrapped_fn(*args, **kwargs):
 
 def generate_random_string(str_len=10):
     return "".join(random.choices(string.ascii_uppercase, k=str_len))
+
+
+# from projects.modular_llm.src.utils.evaluators import Evaluator
+def get_loss(model, evaluator: "Evaluator", **kwargs):
```
Comment (sordonia) [inline mttl/utils.py:204]: same , maybe move somewhere else? in evaluator?

**Review by sordonia — APPROVED** [review]

File: mttl/models/expert_trainer.py (hunk 1 above)
Comment (matheper) [reply mttl/models/expert_trainer.py:17]: duplicated from `mttl.models.utils`

File: mttl/utils.py (hunk 3 above)
Comment (matheper) [reply mttl/utils.py:204]: actually, not used anywhere

**Pull request merged**